encoding/dachunk.go

```diff
@@ -0,0 +1,161 @@
+package encoding
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/core/types"
+	"github.com/scroll-tech/go-ethereum/crypto"
+)
+
+// DAChunk groups consecutive DABlocks with their transactions.
+type DAChunkV0 struct {
+	Blocks       []*DABlock
+	Transactions [][]*types.TransactionData
+}
```

Comment:
all fields of any type that we're expecting to handle solely via interface should be private and exposed via interface, otherwise they're not accessible without casting into the respective type

Reply:
fixed in [1f9facd](https://github.com/scroll-tech/da-codec/pull/25/commits/1f9facd82a2f741013734bae299aab58847b88df).